### ISSUE TITLE
nordic: restarting BLE advertising causes USB disconnect/reconnect or hard reset

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#9442]
While using this test program from #9440, ctrl-C'ing the program and then typing ctrl-D causes a disconnect/reconnect on USB serial. Commenting out the `ble.start_advertising()` prevents the problem. If you comment out the code in `while ble.connected:`, the problem remains.

```py
from adafruit_ble import BLERadio
from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
from adafruit_ble.services.nordic import UARTService
import time

time.sleep(2)
print("---Peripheral---")

ble = BLERadio()
uart = UARTService()
advertisement = ProvideServicesAdvertisement(uart)

#time.sleep(1)
while True:
    print("Advertising...")
    ble.start_advertising(advertisement)  # <--- commenting this out causes the problem to go away.
    while not ble.connected:
        pass
    print("connected")
    while ble.connected:
        if uart.in_waiting:
            print("got", uart.read(uart.in_waiting))
        uart.write(str(time.monotonic()).encode())
        time.sleep(2)
    print("disconnected")
```

**Comment by tannewt** [comment]
Does the board come back up in safe mode? Or is it just USB that is reset?

**Comment by dhalbert** [comment]
It does not come back in safe mode. It looks like this:
```
[15:58:35.446] tio v2.7
[15:58:35.446] Press ctrl-t q to quit
[15:58:35.446] Warning: Could not open tty device (No such file or directory)
[15:58:35.446] Waiting for tty device..
[15:58:43.551] Connected
    [***typed ctrl-C***]
Traceback (most recent call last):
  File "code.py", line 17, in <module>
KeyboardInterrupt: 

Code done running.

Press any key to enter the REPL. Use CTRL-D to reload.

Adafruit CircuitPython 9.1.0 on 2024-07-10; Adafruit ItsyBitsy nRF52840 Express with nRF52840
>>> 
[15:58:50.265] Disconnected
[15:58:51.266] Warning: Could not open tty device (No such file or directory)
[15:58:51.266] Waiting for tty device..
[15:59:09.489] Connected
    [***typed ctrl-C***]
Traceback (most recent call last):
  File "code.py", line 17, in <module>
KeyboardInterrupt: 

Code done running.

Press any key to enter the REPL. Use CTRL-D to reload.

Adafruit CircuitPython 9.1.0 on 2024-07-10; Adafruit ItsyBitsy nRF52840 Express with nRF52840
>>> 
```

**Comment by tannewt** [comment]
Weird! You could use UART debugging to try and sort it out.

**Comment by dhalbert** [comment]
I tried this on a CLUE, and could not reproduce. I uploaded its bootloader to 0.9.3, still could not reproduce. Then I did `_bleio.adapter.erase_bonding()`, and now I can reproduce. So something about having totally empty bonding is causing the problem.

The display on the CLUE shows that it is doing a hard restart.

**Comment by dhalbert** [comment]
Fixed by #9453.
